tic_tac_toe.rb

```diff
@@ -0,0 +1,89 @@
+class Grid
+  attr_accessor :rows
+  attr_reader :columns, :line
+
+  def initialize
+    @columns = '  A    B   C '
```

Comment:
What you store in `@columns`, `@line`, as well as the values of `edge`, `middleA` and `middleB` nested in `@rows` would all be considered parts of the presentation (or rendering) of a grid, and therefore should not be part of the `Grid` class.

Simplest acceptable solution would be to have a `def render` class, which would implement all of this logic, while taking the grid state from `@grid` variable.

Reply:
Additionally on this topic, one of your jobs as a programmer is to understand what is the *state* of your program, and how does it change over time. *state* is basically anything that changes at a given scope, during the lifetime of your program. Variables I pointed out above are not, really, the state of your application. They are helper values for rendering, state local to your incoming `#render` method.

However, your main loop has 2 steps: first player makes move, then second player makes move. This is hidden state, representing "who makes the move right now?", or something like `@active_player`. 
If you make this into an instance variable on `Game` class, your code will get much cleaner.

Good practice is to minimize state, because managing lot of state is difficult. You could follow these broad ideas:
  - state with smaller scope is better, for example instance variables are better then global variables, and local variables are better than instance variables
  - no state is even better, if you can replace having a variable with having an argument to a method it tends to be a good idea.
  
  Those are just heuristics, of course, you have to judge on a case-by-case basis. You could google for some articles about "understanding state in programming".